lib/components/artist/artworks/index.js

```diff
@@ -38,31 +41,67 @@ class Artworks extends React.Component {
         otherWorks.push(this.renderSection('Other Works',
                         other_count,
                         this.props.artist.not_for_sale_artworks,
-                        (artistID, page) => artworksQuery(artistID, 'IS_NOT_FOR_SALE', page)))
+                        () => { return ({ availability: 'IS_NOT_FOR_SALE'}) },
+                        this.resolveQuery,
+                        null))
       }
       return (
         <View style={styles.section}>
           {this.renderSection('Works for Sale',
                               for_sale_count,
                               this.props.artist.for_sale_artworks,
-                              (artistID, page) => artworksQuery(artistID, 'IS_FOR_SALE', page),
-                              () => { this.setState({ completedForSaleWorks: true }) })}
+                              () => { return ({ availability: 'IS_FOR_SALE'}) },
+                              this.resolveQuery,
+                              () => { this.setState({ completedForSaleWorks: true }) }
+          )}
           {otherWorks}
         </View>
       )
     }
   }
 
-  renderSection(title, count, artworks, query, onComplete) {
+  renderSection(title: string, count: number, artworks: any[], stateQuery: any, pageQuery: any, onComplete: any){
     return (
       <View key={title}>
         <SerifText style={styles.heading}>
           <SerifText style={styles.text}>{title}</SerifText> <SerifText style={[styles.text, styles.count]}>({count})</SerifText>
         </SerifText>
-        <ArtworksGrid artworks={artworks} artworksQuery={query} onComplete={onComplete} />
+        <ArtworksGrid artworks={artworks} queryForState={stateQuery} queryForPage={pageQuery} onComplete={onComplete} queryArtworksKeypath="artist" />
       </View>
     )
   }
+
+  resolveQuery = (component: ArtworksGrid, page: number, state: any) : string => {
+    // The page + 1 is to take into account the fact that we _start_ with results already
+    return Artworks.artworksQuery(this.props.relay.route.params.artistID, state.availability, component.state.page + 1)
+  }
```

Comment:
whereas other query resolving _may_ not be able to grab the first page via graphQL